rs/p2p/quic_transport/src/connection_manager.rs

```diff
@@ -415,23 +409,19 @@ impl ConnectionManager {
 
         // Remove peer connections that are not part of subnet anymore.
         // Also remove peer connections that have closed connections.
-        let mut peer_map = self.peer_map.write().unwrap();
-        peer_map.retain(|peer_id, conn_handle| {
+        let peer_map = self.peer_map.read().unwrap();
+        peer_map.iter().for_each(|(peer_id, conn_handle)| {
```

Comment:
So for my understanding, now when a node is not part of the subnet anymore, this will cause a connection drop, which will cause the active_connection to complete, which will in the end cause the peer_map to remove that peer, right? Meaning we don't need to remove it in advance here.

Reply:
Exactly.